staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret.go

```diff
@@ -370,25 +370,31 @@ func handleSecretFromFileSources(secret *corev1.Secret, fileSources []string) er
 	return nil
 }
 
-// handleSecretFromEnvFileSource adds the specified env file source information
+// handleSecretFromEnvFileSources adds the specified env files source information
 // into the provided secret
-func handleSecretFromEnvFileSource(secret *corev1.Secret, envFileSource string) error {
-	fileInfo, err := os.Stat(envFileSource)
-	if err != nil {
-		switch err := err.(type) {
-		case *os.PathError:
-			return fmt.Errorf("error reading %s: %v", envFileSource, err.Err)
-		default:
-			return fmt.Errorf("error reading %s: %v", envFileSource, err)
+func handleSecretFromEnvFileSources(secret *corev1.Secret, envFileSources []string) error {
+	for _, envFileSource := range envFileSources {
+		info, err := os.Stat(envFileSource)
+		if err != nil {
+			switch err := err.(type) {
+			case *os.PathError:
+				return fmt.Errorf("error reading %s: %v", envFileSource, err.Err)
+			default:
+				return fmt.Errorf("error reading %s: %v", envFileSource, err)
+			}
+		}
+		if info.IsDir() {
+			return fmt.Errorf("env secret file cannot be a directory")
+		}
+		err = cmdutil.AddFromEnvFile(envFileSource, func(key, value string) error {
+			return addKeyFromLiteralToSecret(secret, key, []byte(value))
+		})
```

Comment:
What do you do if you have duplicate keys? Is this properly handled by this function? Specifically if they are coming from different files with the same name in different directories?

Reply:
@eddiezane actually asked the same questions 😄 and @KnVerey answered it over here (https://github.com/kubernetes/kubernetes/pull/101646#issuecomment-8827883522)